common/frame.go

```diff
@@ -1270,31 +1270,45 @@ func (f *Frame) isDisabled(selector string, opts *FrameIsDisabledOptions) (bool,
 	return bv, nil
 }
 
+// IsHidden returns true if the first element that matches the selector
+// is hidden. Otherwise, returns false.
 func (f *Frame) IsHidden(selector string, opts goja.Value) bool {
 	f.log.Debugf("Frame:IsHidden", "fid:%s furl:%q sel:%q", f.ID(), f.URL(), selector)
 
-	parsedOpts := NewFrameIsHiddenOptions(f.defaultTimeout())
-	if err := parsedOpts.Parse(f.ctx, opts); err != nil {
+	popts := NewFrameIsHiddenOptions(f.defaultTimeout())
+	if err := popts.Parse(f.ctx, opts); err != nil {
+		k6ext.Panic(f.ctx, "%w", err)
+	}
+	hidden, err := f.isHidden(selector, popts)
+	if err != nil {
 		k6ext.Panic(f.ctx, "%w", err)
 	}
 
-	fn := func(apiCtx context.Context, handle *ElementHandle) (interface{}, error) {
-		value, err := handle.isHidden(apiCtx, 0) // Zero timeout when checking state
-		if err == ErrTimedOut {                  // We don't care about timeout errors here!
-			return value, nil
+	return hidden
+}
+
+func (f *Frame) isHidden(selector string, opts *FrameIsHiddenOptions) (bool, error) {
+	isHidden := func(apiCtx context.Context, handle *ElementHandle) (interface{}, error) {
+		v, err := handle.isHidden(apiCtx, 0) // Zero timeout when checking state
+		if errors.Is(err, ErrTimedOut) {     // We don't care about timeout errors here!
+			return v, nil
 		}
-		return value, err
+		return v, err
 	}
-	actFn := f.newAction(
-		selector, DOMElementStateAttached, parsedOpts.Strict, fn, []string{}, false, true, parsedOpts.Timeout,
+	act := f.newAction(
+		selector, DOMElementStateAttached, opts.Strict, isHidden, []string{}, false, true, opts.Timeout,
 	)
-	value, err := callApiWithTimeout(f.ctx, actFn, parsedOpts.Timeout)
+	v, err := callApiWithTimeout(f.ctx, act, opts.Timeout)
 	if err != nil {
```

Comment:
Should we catch the context timeout and return `false, nil` like we above on line 1293?

Reply:
No, we don't need to because the `callApiWithTimeout` already does that for us.